src/node/building_blocks.ml

```diff
@@ -102,18 +108,38 @@ let should_start_new_epoch last_state_root_update current_time =
     \    The block producer uses this function to determine when to send a\n\
     \    block with an updated state root hash.\n"]
 
+(** Can only included a tezos operation if enough time has already elapsed *)
+let can_included_tezos_operation ~current_time pending_operation =
+  current_time -. pending_operation.Node.requested_at
+  > minimum_waiting_period_for_tezos_operation
+
 let produce_block state =
+  let current_time = Unix.time () in
   let start_new_epoch =
     should_start_new_epoch state.Node.protocol.last_state_root_update
-      (Unix.time ()) in
+      current_time in
   let next_state_root_hash =
     if start_new_epoch then
       let%some snapshot = Snapshots.get_next_snapshot state.snapshots in
       Some snapshot.hash
     else
       None in
+  let operations =
+    List.filter_map
+      (fun pending_operation ->
+        let operation = pending_operation.Node.operation in
+        match operation with
+        | Operation.Core_tezos _ ->
+          if can_included_tezos_operation ~current_time pending_operation then
```

Comment:
What about changing `can_include_tezos_operation` to return `Some operation` or `None`, and here get rid of the `if` statement?

Reply:
Hum, almost agreed. I still feel weird to see:
```ocaml
if can_included_tezos_operation ~current_time pending_operation then
    Some operation
  else
    None
```
within this function. Any argument to help me appreciate this?